packages/fetchai/skills/generic_buyer/handlers.py

```diff
@@ -215,9 +216,11 @@ def _handle_match_accept(
             ledger_api_dialogue = cast(LedgerApiDialogue, ledger_api_dialogue)
             ledger_api_dialogue.associated_fipa_dialogue = fipa_dialogue
             fipa_dialogue.associated_ledger_api_dialogue = ledger_api_dialogue
-            self.context.outbox.put_message(message=ledger_api_msg)
-            self.context.logger.info(
```

Comment:
We now properly submit them in order; waiting till the previous one is finished or timed-out